### PR TITLE
Trace Interface V2

### PR DESCRIPTION
**Pull request by sragss**
I gave up after trying to merge V1 with `moodlezoup/jolt` and decided to write it fresh. Changes are the same. 

I've incorporated feedback from V1.

### CODE REVIEW

File: jolt-core/src/jolt/trace/rv.rs [hunk 1]
```diff
@@ -0,0 +1,738 @@
...
+    // 7: Instruction adds operands (ie, and uses the ADD lookup table)
+    // 8: Instruction subtracts operands
+    // 9: Instruction multiplies operands
+    // 10: Instruction involves non-deterministic advice?
+    // 11: Instruction asserts lookup output as false
+    // 12: Instruction asserts lookup output as true
+    // 13: Sign-bit of imm
+    // 14: Instruction is lui
+
+    let flag_0 = match self.opcode {
```
Comment (moodlezoup) [inline jolt-core/src/jolt/trace/rv.rs:545]: nit: 
```suggestion
    let mut flags = vec![F::zero(); 15];
    flags[0] = ... 
    // etc.
```

**Review by moodlezoup — COMMENTED** [review]
some cosmetic nits

File: jolt-core/src/jolt/trace/mod.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+}
+
+impl MemoryOp {
+    fn new_read(address: u64, value: u64) -> Self {
+        Self::Read(address, value)
+    }
+
+    fn new_write(address: u64, old_value: u64, new_value: u64) -> Self {
+        Self::Write(address, old_value, new_value)
+    }
```
[the remark this answers is not in the document]
Comment (sragss) [reply jolt-core/src/jolt/trace/mod.rs:19]: Good catch. Done.

File: jolt-core/src/jolt/trace/rv.rs [hunk 3]
```diff
@@ -0,0 +1,738 @@
...
+    let flag_11 = match self.opcode {
+      _ => false,
+    };
+
+    // not incorporating assert false instructions yet
+    let flag_12 = match self.opcode {
+      _ => false,
+    };
+
+    // not incorporating advice instructions yet
```
[the remark this answers is not in the document]
Comment (sragss) [reply jolt-core/src/jolt/trace/rv.rs:635]: Done.

File: jolt-core/src/jolt/instruction/add.rs [hunk 4]
```diff
@@ -11,15 +11,17 @@ use crate::utils::instruction_utils::{
...
 
 #[derive(Copy, Clone, Default, Debug)]
-pub struct ADDInstruction(pub u64, pub u64);
+pub struct ADDInstruction<const W: usize>(pub u64, pub u64);
 
-impl JoltInstruction for ADDInstruction {
+pub type ADD32Instruction = ADDInstruction<32>;
+pub type ADD64Instruction = ADDInstruction<64>;
+
+impl<const W: usize> JoltInstruction for ADDInstruction<W> {
```
[the remark this answers is not in the document]
Comment (sragss) [reply jolt-core/src/jolt/instruction/add.rs:19]: Done.

File: common/src/path.rs [hunk 5]
```diff
@@ -0,0 +1,30 @@
+#[allow(non_snake_case)]
+/// For clean importing
+pub mod JoltPaths {
+    use std::{path::PathBuf, env};
+
+    // Traces:    <target dir>/<TRACE_DIR_NAME>/<program name>.<TRACE_FILE_SUFFIX>
+    // Programs:  <target dir>/<TRACE_DIR_NAME>/<program name>.<ELF_TRACE_FILE_SUFFIX>
+    const TRACE_DIR_NAME: &'static str = "traces";
+    const TRACE_FILE_SUFFIX: &'static str = "jolt";
+    const ELF_TRACE_FILE_SUFFIX: &'static str = "joltprogram";
```
[the remark this answers is not in the document]
Comment (sragss) [reply common/src/path.rs:10]: Done.

**Pull request merged**